test/integration/scheduler/preemption_test.go

```diff
@@ -289,6 +302,60 @@ var _ = ginkgo.Describe("Preemption", func() {
 			util.ExpectWorkloadsToBeAdmitted(ctx, k8sClient, alphaCQ.Name, alphaHighWl1)
 			util.ExpectWorkloadsToBeAdmitted(ctx, k8sClient, betaCQ.Name, betaLowWl)
 		})
+
+		ginkgo.It("Should preempt all necessary workloads in concurrent scheduling", func() {
+			ginkgo.By("Creating workloads in beta-cq that borrow quota")
+
+			betaMidWl := testing.MakeWorkload("beta-mid", ns.Name).
+				Queue(betaQ.Name).
+				Priority(midPriority).
+				Request(corev1.ResourceCPU, "3").
+				Obj()
+			gomega.Expect(k8sClient.Create(ctx, betaMidWl)).To(gomega.Succeed())
+			betaHighWl := testing.MakeWorkload("beta-high", ns.Name).
+				Queue(betaQ.Name).
+				Priority(highPriority).
+				Request(corev1.ResourceCPU, "3").
+				Obj()
+			gomega.Expect(k8sClient.Create(ctx, betaHighWl)).To(gomega.Succeed())
+
+			util.ExpectWorkloadsToBeAdmitted(ctx, k8sClient, betaCQ.Name, betaMidWl, betaHighWl)
+
+			ginkgo.By("Creating workload in alpha-cq and game-cq that need to preempt")
+			alphaMidWl := testing.MakeWorkload("alpha-mid", ns.Name).
+				Queue(alphaQ.Name).
+				Priority(midPriority).
+				Request(corev1.ResourceCPU, "2").
+				Obj()
+
+			gamaMidWl := testing.MakeWorkload("gama-mid", ns.Name).
+				Queue(gamaQ.Name).
+				Priority(midPriority).
+				Request(corev1.ResourceCPU, "2").
+				Obj()
+
+			gomega.Expect(k8sClient.Create(ctx, alphaMidWl)).To(gomega.Succeed())
+			gomega.Expect(k8sClient.Create(ctx, gamaMidWl)).To(gomega.Succeed())
+
+			// since the two pending workloads are not aware of each other both of them
+			// will request the eviction of betaMidWl only
+			util.FinishEvictionForWorkloads(ctx, k8sClient, betaMidWl)
+
+			// one of alpha-mid and gama-mid should be admitted
+			var firstAdmitted []*kueue.Workload
+			gomega.Eventually(func() []*kueue.Workload {
+				firstAdmitted = util.FilterAdmittedWorkloads(ctx, k8sClient, alphaMidWl, gamaMidWl)
+				return firstAdmitted
+			}, util.Timeout, util.Interval).Should(gomega.HaveLen(1))
+			// and stay that way
+			gomega.Consistently(util.FilterAdmittedWorkloads(ctx, k8sClient, alphaMidWl, gamaMidWl), util.ConsistentDuration, util.Interval).Should(gomega.Equal(firstAdmitted))
```

Comment:
we shouldn't need this check. We should avoid calling `Consistently` as it's pretty much a busy wait.

We can instead confirm that both workloads are admitted at the end, and verify that the preempted workloads are not admitted anymore.